internal/resources/business_unit/resource.go

```diff
@@ -0,0 +1,934 @@
+						},
+						"salutation": schema.StringAttribute{
+							Description: "Salutation of the contact, for example Ms., Mr.",
+							Optional:    true,
+						},
+						"first_name": schema.StringAttribute{
+							Description: "First name of the contact",
+							Optional:    true,
+						},
+						"last_name": schema.StringAttribute{
+							Description: "Last name of the contact",
+							Optional:    true,
+						},
+						"street_name": schema.StringAttribute{
+							Description: "Name of the street",
+							Optional:    true,
+						},
+						"street_number": schema.StringAttribute{
+							Description: "Street number",
+							Optional:    true,
+						},
+						"additional_street_info": schema.StringAttribute{
+							Description: "Further information on the street address",
+							Optional:    true,
+						},
+						"postal_code": schema.StringAttribute{
+							Description: "Postal code",
+							Optional:    true,
+						},
+						"city": schema.StringAttribute{
+							Description: "Name of the city",
+							Optional:    true,
+						},
+						"region": schema.StringAttribute{
+							Description: "Name of the region",
+							Optional:    true,
+						},
+						"state": schema.StringAttribute{
+							Description: "Name of the state",
+							Optional:    true,
+						},
+						"company": schema.StringAttribute{
+							Description: "Name of the company",
+							Optional:    true,
+						},
+						"department": schema.StringAttribute{
+							Description: "Name of the department",
+							Optional:    true,
+						},
+						"building": schema.StringAttribute{
+							Description: "Name or number of the building",
+							Optional:    true,
+						},
+						"apartment": schema.StringAttribute{
+							Description: "Name or number of the apartment",
+							Optional:    true,
+						},
+						"po_box": schema.StringAttribute{
+							Description: "Post office box number",
+							Optional:    true,
+						},
+						"phone": schema.StringAttribute{
+							Description: "Phone number",
+							Optional:    true,
+						},
+						"mobile": schema.StringAttribute{
+							Description: "Mobile phone number",
+							Optional:    true,
+						},
+						"email": schema.StringAttribute{
+							Description: "Email address",
+							Optional:    true,
+						},
+						"fax": schema.StringAttribute{
+							Description: "Fax number",
+							Optional:    true,
+						},
+						"additional_address_info": schema.StringAttribute{
+							Description: "Further information on the Address",
+							Optional:    true,
+						},
+					},
+				},
+			},
+			"store": schema.ListNestedBlock{
+				Description: "Stores that are part of the Business Unit.",
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"key": schema.StringAttribute{
+							Description: "User-defined unique identifier of the Store",
+							Required:    true,
+						},
+						"type_id": schema.StringAttribute{
```

Comment:
This one should not be required, as it can onlye be a single type (Store). We should however add id as an optional reference instead, and make both id and key optional, to fit closer to the API